Custom query command (#158)
mramendi/kie-docs: doc-content/enterprise-only/admin-and-config/custom-query-proc.adoc

[id='custom-query-proc']
= Running a custom query on the {PRODUCT} database

You can use the `ExecuteSQLQueryCommand` executor command to run a custom query on the {PRODUCT} database, including the audit log data tables. You can set up a job that runs this command in {CENTRAL}.

.Procedure
. In {CENTRAL}, go to *Manage* > *Jobs*.
. Click *New Job*.

> . Enter values for *Business Key*, *Due On*, and *Retries* fields.

Reviewer: ```suggestion
. In the *Business Key* field, enter a value.
. In the *Due On* field, enter a value.
. In the *Retries* field, enter a value.
```
See [field](https://www.ibm.com/docs/en/ibm-style?topic=elements-ui#field)

> . Enter the following command into the *Type* field.

Reviewer: ```suggestion
. In the *Type* field, enter the following command:
```
See [field](https://www.ibm.com/docs/en/ibm-style?topic=elements-ui#field)

+
[source]
----
org.jbpm.executor.commands.ExecuteSQLQueryCommand
----
+
. To configure parameters, complete the following steps:
+
--

> .. Open the *Advanced* tab.

Reviewer: ```suggestion
.. Click the *Advanced* tab.
```
See [tab](https://www.ibm.com/docs/en/ibm-style?topic=elements-ui#tab)

.. Click *Add Parameter*.

> .. Enter a parameter in the *Key* column and enter a parameter value in the *Value* column.

Reviewer: ```suggestion
.. In the *Key* column, enter a parameter value.
.. In the *Value* column, enter a parameter value.
```
See [field](https://www.ibm.com/docs/en/ibm-style?topic=elements-ui#field).

--
+
For the list of parameters for the command, see xref:custom-query-params-ref[].
+
. Click *Create*. {CENTRAL} creates the custom query job.
. Optional: If you want to retrieve the results of the query, complete the following steps:

> .. In the list of jobs that {CENTRAL} displays, find the job that you started. If the job is not present in the list, remove any filters in the *Active filters* list.

Reviewer: ```suggestion
.. In the list of jobs that {CENTRAL} displays, find the job that you started. If the job is not present in the list, remove any filters from the *Active filters* list.
```

.. Record the *id* value for the job.
.. Using a web browser, access the Swagger documentation on your {KIE_SERVER} at `\<kie_server_address>/docs`, for example, `http://localhost:8080/kie-server/docs/`.
.. Click the `GET /server/jobs/{jobId}` request.

> .. In the *jobId* parameter field, enter the *id* value that you recorded.

Reviewer: ```suggestion
.. In the *jobId* field, enter the *id* value that you recorded.
```
Removed "parameter" descriptors field for consistency with other fields.

> .. Set the *withErrors* parameter to `true`.

Reviewer: What type of Ui element is the parameter. If it is a radio button or list, follow the [radio button style](https://www.ibm.com/docs/en/ibm-style?topic=elements-ui#radio_button) or [list style](https://www.ibm.com/docs/en/ibm-style?topic=elements-ui#list).

> .. Set the *withData* parameter to `true`.

Reviewer: What type of Ui element is the parameter. If it is a radio button or list, follow the [radio button style](https://www.ibm.com/docs/en/ibm-style?topic=elements-ui#radio_button) or [list style](https://www.ibm.com/docs/en/ibm-style?topic=elements-ui#list).

.. Click *Execute*.

> .. Review the *Server response* field. If the SQL query succeeded, the result is under the `"response-data"` key.

Reviewer: ```suggestion
.. Review the *Server response* field. If the SQL query succeeded, the result is under the `"response-data"` key.
```
If the query fails, does anything display under the `"response-data"` key as well?
